core/iwasm/interpreter/wasm_interp_classic.c

```diff
@@ -36,7 +36,7 @@ typedef float64 CellType_F64;
  * multi-threading mode since it may be changed by other
  * threads in memory.grow
  */
-#define get_linear_mem_size() memory->memory_data_size
+#define get_linear_mem_size() GET_LINEAR_MEMORY_SIZE(memory)
```

Comment:
do we still need `get_linear_mem_size` macro given that `GET_LINEAR_MEMORY_SIZE` will work for both shared and non-shared memory (as per @wenyongh 's comment https://github.com/bytecodealliance/wasm-micro-runtime/pull/2834/files#r1408616821)? Could we just use `GET_LINEAR_MEMORY_SIZE` ?

Reply:
I'll do that. 

What about AOT instead? I started looking into the code but not very familiar with that part, any hints on where to look to apply similar changes to what I did for classic and fast interpreter?

Reply:
Sorry I checked the code again in classic interpreter, we had better keep the macro `get_linear_mem_size`, its meaning is different from the macro GET_LINEAR_MEMORY_SIZE: it uses local variable linear_mem_size in sing-threading mode, as linear_mem_size is only updated after function calls and memory.grow, we may improve the performance in interpreter mode. If we change to GET_LINEAR_MEMORY_SIZE, it will get the memory size from memory structure each time, which may impacts the performance in single-threading mode.

Reply:
> Sorry I checked the code again in classic interpreter, we had better keep the macro `get_linear_mem_size`, its meaning is different from the macro GET_LINEAR_MEMORY_SIZE: it uses local variable linear_mem_size in sing-threading mode, as linear_mem_size is only updated after function calls and memory.grow, we may improve the performance in interpreter mode. If we change to GET_LINEAR_MEMORY_SIZE, it will get the memory size from memory structure each time, which may impacts the performance in single-threading mode.

The nuttx CI failure should be caused by the changing.

Reply:
> I'll do that.
> 
> What about AOT instead? I started looking into the code but not very familiar with that part, any hints on where to look to apply similar changes to what I did for classic and fast interpreter?

The AOT mode doesn't use linear memory size to do boundary check for normal load/store opcodes, instead it uses the `mem_bound_check_nbytes` fields:
https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/core/iwasm/compilation/aot_emit_memory.c#L42-L60

 It only uses linear memory size in checking the size in opcode memory.fill/memory.copy:
https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/core/iwasm/compilation/aot_emit_memory.c#L935-L944

Reply:
Yes, the problem I had was that I can't use `get_linear_mem_size()` for the first initialization in `wasm_interp_call_func_bytecode` because otherwise it uses `#define get_linear_mem_size() linear_mem_size`. Instead we need to get `memory->memory_data_size` at the beginning, afterwards we can rely on `get_linear_mem_size()` to get the `linear_mem_size` cached value.